go/vt/vtgate/planbuilder/operators/update.go

```diff
@@ -926,3 +998,113 @@ func nullSafeNotInComparison(ctx *plancontext.PlanningContext, updatedTable *vin
 
 	return finalExpr
 }
+
+func buildChangedVindexesValues(
+	ctx *plancontext.PlanningContext,
+	update *sqlparser.Update,
+	table *vindexes.Table,
+	ksidCols []sqlparser.IdentifierCI,
+	assignments []SetExpr,
+) (vv map[string]*engine.VindexValues, ownedVindexQuery *sqlparser.Select, subQueriesArgOnChangedVindex []string) {
+	changedVindexes := make(map[string]*engine.VindexValues)
+	selExprs, offset := initialQuery(ksidCols, table)
+	for i, vindex := range table.ColumnVindexes {
+		vindexValueMap := make(map[string]evalengine.Expr)
+		var compExprs []sqlparser.Expr
+		for _, vcol := range vindex.Columns {
+			subQueriesArgOnChangedVindex, compExprs =
+				createAssignmentExpressions(ctx, assignments, vcol, subQueriesArgOnChangedVindex, vindexValueMap, compExprs)
+		}
+		if len(vindexValueMap) == 0 {
+			// Vindex not changing, continue
+			continue
+		}
+		if i == 0 {
+			panic(vterrors.VT12001(fmt.Sprintf("you cannot UPDATE primary vindex columns; invalid update on vindex: %v", vindex.Name)))
+		}
+		if _, ok := vindex.Vindex.(vindexes.Lookup); !ok {
+			panic(vterrors.VT12001(fmt.Sprintf("you can only UPDATE lookup vindexes; invalid update on vindex: %v", vindex.Name)))
+		}
+
+		// Checks done, let's actually add the expressions and the vindex map
+		selExprs = append(selExprs, aeWrap(sqlparser.AndExpressions(compExprs...)))
+		changedVindexes[vindex.Name] = &engine.VindexValues{
+			EvalExprMap: vindexValueMap,
+			Offset:      offset,
+		}
+		offset++
+	}
+	if len(changedVindexes) == 0 {
+		return nil, nil, nil
+	}
+	// generate rest of the owned vindex query.
+	ovq := &sqlparser.Select{
+		SelectExprs: selExprs,
+		OrderBy:     update.OrderBy,
+		Limit:       update.Limit,
+		Lock:        sqlparser.ForUpdateLock,
+	}
+	return changedVindexes, ovq, subQueriesArgOnChangedVindex
```

Comment:
it it not super important but it is a bit confusing why we are naming the return arguments on line `1008` but it seems like there is no reason to not directly used the variable declared in `1008`.

Reply:
fixed